spec/integration/outputs/elasticsearch/node_spec.rb

```diff
@@ -0,0 +1,35 @@
+require "spec/es_spec_helper"
+require "logstash/outputs/elasticsearch/protocol"
+
+describe "elasticsearch node client" do
+  # Test ElasticSearch Node Client
+  # Reference: http://www.elasticsearch.org/guide/reference/modules/discovery/zen/
+
+  it "should support hosts in both string and array" do
+    # Because we defined *hosts* method in NodeClient as private,
+    # we use *obj.send :method,[args...]* to call method *hosts*
+    client = LogStash::Outputs::Elasticsearch::Protocols::NodeClient.new(:host => "localhost")
```

Comment:
this would actually live more in a let variable, or even better as a subject of the test.